### PR TITLE
User Settings

### PR DESCRIPTION
**Pull request by nilsjung**

### CODE REVIEW

**Comment by nilsjung** [comment]
I have done some refactoring stuff here as well: 
* i found a solution to introduce **private routes**. So the redirect, if not authenticated, is much more easier. as a result the props are now generated all in the chat-component with `mapStateToProps`.
*  i splitted the `helperActions` to have a clearer structure.
*  The **Input-Component** is now a uncontrolled Component. Means the state is not in the Input-Component but in the Component that calls the Input. This was necessary to inject a default value to the Input field for the Settiings.

File: app/src/actions/userActions.js [hunk 1]
```diff
@@ -106,6 +105,26 @@ export function login({ email, password }) {
...
+    return (dispatch) => {
+        dispatch(isLoading(true));
+        request
+            .post(userEndpoint + user._id)
+            .set(signHeader(token))
+            .send(user)
+            .then((result) => {
+                dispatch(isLoading(false));
+                dispatch(isSuccess(true));
+                dispatch(showPopup(result.body.message));
```
Comment (karsil) [inline app/src/actions/userActions.js]: Updated user account information is only visible after after a re-login. Shouldn't it been visible right after submitting the formular in *Settings*?

File: app/src/actions/userActions.js [hunk 2]
```diff
@@ -136,6 +155,13 @@ export function userLogin({ user, accessToken, chatrooms, userchats }) {
     };
 }
 
+export function userUpdate(user) {
+    return {
+        type: USER_UPDATE,
+        user,
+    };
+}
+
```
Comment (karsil) [inline app/src/actions/userActions.js]: Is this action being used anywhere? Could not find any use...

File: app/src/actions/userActions.js [hunk 3]
```diff
@@ -106,6 +105,26 @@ export function login({ email, password }) {
     };
 }
 
+export function updateAction(user, token) {
```
Comment (karsil) [inline app/src/actions/userActions.js]: Missing comment, what this really does, *updateAction* is very generic. Suggestion: updateUserProfileAction

File: app/src/components/registration/Registration.js [hunk 4]
```diff
@@ -34,7 +34,7 @@ function mapDispatchToProps(dispatch) {
     return {
         registerUser: (user) => {
             dispatch(registerActions.registerUser(user));
-            dispatch(helperActions.isSuccess(null)); // reset the success status
+            dispatch(isSuccess(null)); // reset the success status
```
Comment (karsil) [inline app/src/components/registration/Registration.js]: For clearification: Why is this necessary, as `dispatch(isSuccess(true))` is executed in the action in the line above?

File: app/src/components/user/Settings.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+    onSubmit = (event) => {
+
+        event.preventDefault();
+        this.props.updateUser(this.state, this.props.authToken);
+    };
+
+    onChange = (key) => (value) => {
+        this.setState({
+            [key]: value,
```
Comment (karsil) [inline app/src/components/user/Settings.js]: Clever!

File: app/src/actions/userActions.js (hunk 1 above)
Comment (nilsjung) [reply app/src/actions/userActions.js]: i get a message after updating...

File: app/src/components/registration/Registration.js (hunk 4 above)
Comment (nilsjung) [reply app/src/components/registration/Registration.js]: it is not. seems to be legacy-code ;)

File: app/src/actions/userActions.js (hunk 2 above)
Comment (nilsjung) [reply app/src/actions/userActions.js]: no you are right. its not. its a result of a first thought, that it could be useful. but i'll delete it for the moment...

File: app/src/components/mixins/PrivateRoute.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        <Route
+            {...attributes}
+            render={(props) =>
+                authed === true ? (
+                    <Component {...props} />
+                ) : (
+                    <Redirect
+                        to={{
+                            pathname: '/login',
+                            state: { from: props.location },
```
[the remark this answers is not in the document]
Comment (nilsjung) [reply app/src/components/mixins/PrivateRoute.js]: not for now. i remove it for the moment. it could be necessary when we want to redirect to the former position...

File: app/src/actions/userActions.js (hunk 3 above)
Comment (nilsjung) [reply app/src/actions/userActions.js]: i will change the action name...

**Comment by nilsjung** [comment]
@karsil fixed some stuff

File: app/src/actions/userActions.js (hunk 1 above)
Comment (karsil) [reply app/src/actions/userActions.js]: I don’t mean the pop-up, but the first & last name on top of the page.

Comment (nilsjung) [reply app/src/actions/userActions.js]: ah i see... this is what the `userUpdate` function was supposed to do.  :D

Comment (nilsjung) [reply app/src/actions/userActions.js]: fixed the store-update
